components/eam/src/chemistry/modal_aero/modal_aero_convproc.F90

```diff
@@ -1786,12 +1468,556 @@ subroutine ma_convproc_tend(                                           &
+
+
+   integer      :: kk           ! vertical index
+   integer      :: kp1, km1, kp1x, km1x ! index of kk+1 and kk-1. x: with bounds
+   integer      :: icnst        ! index of pcnst_extd
+   real(r8)     :: fa_u_dp      ! fa_u * dp at the current level [mb]
+   real(r8)     :: fluxin, fluxout, netflux   ! in, out and net flux [kg/kg/s * mb]
+   real(r8)     :: tmpa         ! working variable of mass flux [mb/s]
+   real(r8)     :: netsrce      ! working variable of flux source [kg/kg/s * mb]
+   real(r8)     :: dconudt_aqchem(pcnst_extd,pverp)  ! d(conu)/dt by aqueous chem [kg/kg/s]
+
+   ! initialize variables
+   sumactiva(:) = 0.0
+   sumaqchem(:) = 0.0
+   sumwetdep(:) = 0.0
+   dcondt(:,:) = 0.0
+   dcondt_wetdep(:,:) = 0.0
+   dconudt_aqchem(:,:) = 0.0    ! aqueous chemistry is ignored in current code
+
+   ! loop from ktop to kbot
+   do kk = ktop, kbot
+      kp1 = kk+1
+      km1 = kk-1
+      kp1x = min( kp1, pver )
+      km1x = max( km1, 1 )
+      fa_u_dp = fa_u(kk)*dpdry_i(kk)
+      do icnst = 2, pcnst_extd
+         if (doconvproc_extd(icnst)) then
+            ! First compute fluxes using environment subsidence/lifting and
+            ! entrainment/detrainment into up/downdrafts,
+            ! to provide an additional mass balance check
+            ! (this could be deleted after the code is well tested)
+            fluxin  = mu_i(kk )*min(chat(icnst,kk ),const(icnst,km1x))     &
+                    - md_i(kp1)*min(chat(icnst,kp1),const(icnst,kp1x))     &
+                    + dudp(kk)*conu(icnst,kk) + dddp(kk)*cond(icnst,kp1)
+            fluxout = mu_i(kp1)*min(chat(icnst,kp1),const(icnst,kk))       &
+                    - md_i(kk )*min(chat(icnst,kk ),const(icnst,kk))       &
+                    + (eudp(kk) + eddp(kk)) * const(icnst,kk)
+
+            netflux = fluxin - fluxout
+
+            ! Now compute fluxes as in convtran, and also source/sink terms
+            ! (version 3 limit fluxes outside convection to mass in appropriate layer
+            ! (these limiters are probably only safe for positive definite quantitities
+            ! (it assumes that mu and md already satify a courant number limit of 1)
+            if (iflux_method /= 2) then
+               fluxin  =     mu_i(kp1)*conu(icnst,kp1)                          &
+                           + mu_i(kk )*min(chat(icnst,kk ),const(icnst,km1x))   &
+                         - ( md_i(kk )*cond(icnst,kk)                           &
+                           + md_i(kp1)*min(chat(icnst,kp1),const(icnst,kp1x)) )
+               fluxout =     mu_i(kk )*conu(icnst,kk)                           &
+                           + mu_i(kp1)*min(chat(icnst,kp1),const(icnst,kk ))    &
+                         - ( md_i(kp1)*cond(icnst,kp1)                          &
+                           + md_i(kk )*min(chat(icnst,kk ),const(icnst,kk )) )
+            else
```

Comment:
From the code coverage, it looks like the else block is never invoked. Would you please check? Thanks!

Reply:
I will check it and remove if not invoked.